evcxr/src/module.rs

```diff
@@ -317,6 +353,26 @@ fn run_cargo(
     }
 }
 
+/// Process one line from cargo, either copying it to stderr or ignoring.
+///
+/// At this point it looks for messages about compiling dependency crates.
+fn tee_error_line(line: &[u8]) {
+    use std::io::Write;
+    static CRATE_COMPILING: OnceCell<regex::bytes::Regex> = OnceCell::new();
+    let crate_compiling = CRATE_COMPILING
+        .get_or_init(|| regex::bytes::Regex::new("^\\s*Compiling (\\w+)(?:\\s+.*)?$").unwrap());
```

Comment:
@d86leader, it looks like this can fail at runtime. When I tried to install evcxr from GitHub, shortly after starting, it panicked:
```
Welcome to evcxr. For help, type :help
>> thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: Syntax(
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
regex parse error:
    ^\s*Compiling (\w+)(?:\s+.*)?$
     ^^
error: Unicode-aware Perl class not found (make sure the unicode-perl feature is enabled)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
)', evcxr\src\module.rs:363:87
```

Might be necessary to enable the [`unicode-perl` feature](https://docs.rs/regex/1.8.1/regex/#unicode-features) for the `regex` dependency. Or adjust the pattern to not use any of these character classes but instead only something like `[ \t]`, ... (and maybe add a comment then above the pattern warning that no Unicode character classes should be used)

Reply:
Interesting, it's surprising to me that the behaviour is not reliable between systems. What OS do you run it on, and do you do some unusual compilation? I'll try to reproduce and enable the required features.

Reply:
I am using Windows 10, with Cargo 1.68.2 (6feb7c9cf 2023-03-26)

Also, I initially started building with `cargo install evcxr_repl`, then cancelled this and ran `cargo install --force --git https://github.com/evcxr/evcxr.git evcxr_repl`. Not sure if that had any negative effect, but I don't think it should.

Reply:
Thanks for noticing this! I think the reason it's not happening when building from a source checkout is because of the cargo workspace. `evcxr_jupyter` depends on `zeromq`, which depends on `regex` with default features enabled.

Reply:
Ok, makes sense. This patch fixes the problem:
```
diff --git a/evcxr/src/module.rs b/evcxr/src/module.rs
index 2de1105..6e4dbec 100644
--- a/evcxr/src/module.rs
+++ b/evcxr/src/module.rs
@@ -359,8 +359,9 @@ fn run_cargo(
 fn tee_error_line(line: &[u8]) {
     use std::io::Write;
     static CRATE_COMPILING: OnceCell<regex::bytes::Regex> = OnceCell::new();
-    let crate_compiling = CRATE_COMPILING
-        .get_or_init(|| regex::bytes::Regex::new("^\\s*Compiling (\\w+)(?:\\s+.*)?$").unwrap());
+    let crate_compiling = CRATE_COMPILING.get_or_init(|| {
+        regex::bytes::Regex::new("^[ \t]*Compiling ([a-zA-Z0-9_-]+)(?:[ \t]+.*)?$").unwrap()
+    });
     if let Some(captures) = crate_compiling.captures(line) {
         let crate_name = captures.get(1).unwrap().as_bytes();
         if crate_name != CRATE_NAME.as_bytes() {
```
I can send a new pull request if that's easier for you

Reply:
> This patch fixes the problem

@d86leader, might be good to add a comment above the regex though explaining why the character classes (e.g. `\w`) are not used. Otherwise without comment the next person who comes across this code might think the regex can be simplified, and reintroduces this issue.

Reply:
Would also be good to add a build with the specified regex version. That way noone would also be able to create new regexes with the same problem.

Reply:
@d86leader, just to make sure there is no misunderstanding, do you want to submit a pull request (either adjusting the regex pattern or enabling the `unicode-perl` feature), or should I do that?

Reply:
Sorry, I mean to submit the patch above, but forgot, although I'm wondering if we should just enable `unicode-perl`. It doesn't appear to slow down compilation of the regex crate - not compared with enabling all regex features anyway. Happy to receive a PR from anyone for either approach - otherwise I'll make the change myself.

Reply:
If this is only a small change (and if you have tested it already anyway), maybe it would be easiest if you do it yourself, so that you can choose the solution you prefer.

Reply:
Thanks for the prompt. I've pushed a fix